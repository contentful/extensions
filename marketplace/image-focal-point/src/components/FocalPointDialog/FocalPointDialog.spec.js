import React from 'react';
import { render } from '@testing-library/react';

import mockProps from '../../test/mockProps';
import { FocalPointDialog } from './FocalPointDialog';

const props = {
  file: mockProps.file,
  focalPoint: mockProps.focalPoint,
  onClose: jest.fn(),
  onSave: jest.fn(),
  sdk: mockProps.sdk
};

describe('FocalPointDialog', () => {
  it('should render the focal point dialog', () => {
    const { container } = render(<FocalPointDialog {...props} />);
    expect(container).toMatchSnapshot();
  });
});
