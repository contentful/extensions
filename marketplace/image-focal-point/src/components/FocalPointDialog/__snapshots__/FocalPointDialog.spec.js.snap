// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocalPointDialog should render the focal point dialog 1`] = `
<div>
  <div
    class="ModalHeader__ModalHeader___1yD1S"
    data-test-id="cf-ui-modal-header"
  >
    <h1
      class="ModalHeader__ModalHeader__title___3IuOy"
    >
      Set focal point
    </h1>
    <button
      class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
      data-test-id="cf-ui-icon-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
        tabindex="-1"
      >
        <svg
          class="Icon__Icon___38Epv IconButton__IconButton__icon___3yZQN Icon__Icon--small___1yGZK undefined"
          data-test-id="cf-ui-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        <span
          class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
        >
          Close
        </span>
      </span>
    </button>
  </div>
  <div
    class="ModalContent__ModalContent___2mf3h"
    data-test-id="cf-ui-modal-content"
  >
    <div
      class="css-k008qs"
    >
      <div>
        <h2
          class="Subheading__Subheading___2mA9j css-cet0rr"
          data-test-id="cf-ui-subheading"
        >
          Select position of focal point
        </h2>
        <div
          class="css-y836as"
        >
          <div
            class="css-1njnee2"
            role="button"
            tabindex="-1"
          >
            <img
              alt="focal point preview"
              src="https://mycdn.com/some_image.png"
            />
          </div>
        </div>
      </div>
      <div
        class="css-1p7lqom"
      >
        <h2
          class="Subheading__Subheading___2mA9j css-cet0rr"
          data-test-id="cf-ui-subheading"
        >
          Preview for different screen sizes
        </h2>
        <div
          class="css-k008qs"
        >
          <div
            class=""
          >
            <div
              style="width: 410px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 410px; height: 276px; transform: translate3d(0px, -28.72524590163934px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Desktop
            </p>
          </div>
        </div>
        <div
          class="css-k008qs"
        >
          <div
            class=""
          >
            <div
              style="width: 280px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 280px; height: 188px; transform: translate3d(0px, 0px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Tablet
            </p>
          </div>
          <div
            class="css-1f2k2gl"
          >
            <div
              style="width: 120px; height: 180px; overflow: hidden; position: relative;"
            >
              <img
                alt="Preview with cropping applied"
                src="https://mycdn.com/some_image.png"
                style="width: 268px; height: 180px; transform: translate3d(0px, 0px, 0); transition: transform 0.3s cubic-bezier(0.13, 0.62, 0.11, 0.99);"
              />
            </div>
            <p
              class="css-qslnu8"
            >
              Mobile
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ModalControls__ModalControls___2bTQx"
    data-test-id="cf-ui-modal-controls"
  >
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp Button__Button--positive___1t6w1"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Save
        </span>
      </span>
    </button>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp Button__Button--muted___2Wair"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Close
        </span>
      </span>
    </button>
  </div>
</div>
`;
